openmmtools/states.py

```diff
@@ -1834,7 +1834,13 @@ def __init__(self, positions, velocities=None, box_vectors=None):
         self._collective_variables = None
         self._kinetic_energy = None
         self._potential_energy = None
-        self._initialize(copy.deepcopy(positions), copy.deepcopy(velocities), copy.deepcopy(box_vectors))
+        args = []
+        for input in [positions, velocities, box_vectors]:
+            if isinstance(input, unit.Quantity) and isinstance(input._value, list):
```

Comment:
We can expand this and just convert on initialization everything to a numpy array, not just lists. I think Patrick also mentioned this would be useful. You can remove entirely the if-else in this case as doing `np.array()` basically creates a copy.

Reply:
> You can remove entirely the if-else

Or maybe substitute it with
```python
if not isinstance(input, np.ndarray):
    # Convert to numpy array
```
to avoid creating a copy of a big array when you don't need it.

Reply:
The problem is that  applying np.array() to a list of Vec3 with units, it gives an array of quantities:
>>> q = [(2, 4, 5), (8, 9, 5)]*simtk.unit.nanometer
>>> q
Quantity(value=[(2, 4, 5), (8, 9, 5)], unit=nanometer)
>>> print(np.array(q))
[[Quantity(value=2, unit=nanometer) Quantity(value=4, unit=nanometer)
  Quantity(value=5, unit=nanometer)]
 [Quantity(value=8, unit=nanometer) Quantity(value=9, unit=nanometer)
  Quantity(value=5, unit=nanometer)]]

Reply:
Yes, this was causing the problem with the reporter.

Reply:
Sorry! I actually meant this
```python
for input in [positions, velocities, box_vectors]:
    if not isinstance(input._value, np.ndarray):
        args.append(np.array(input/input.unit)*input.unit)
    else:
        args.append(copy.deepcopy(input))
```

Reply:
if input were not a quantity (None, for instance), the "if not" statement would give an attribute error, am I wrong?

Reply:
Yes. The problem was caused by using the mdtraj output of positions (via the function openmm_positions https://github.com/mdtraj/mdtraj/blob/34f77aaf767f27d486e8f4762c08fbfd5d3ea03e/mdtraj/core/trajectory.py#L1182 ) as the input of the SamplerState. That output is a list of Vec3s with units.